haystack/tools/parameters_schema_utils.py

```diff
@@ -47,6 +47,42 @@ def _get_param_descriptions(method: Callable) -> Tuple[str, Dict[str, str]]:
     return parsed_doc.short_description or "", param_descriptions
 
 
+def _get_component_param_descriptions(component: Any) -> Tuple[str, Dict[str, str]]:
+    """
+    Get parameter descriptions from a component, handling both regular Components and SuperComponents.
+
+    For regular components, this extracts descriptions from the run method's docstring.
+    For SuperComponents, this extracts descriptions from the underlying pipeline components.
+
+    :param component: The component to extract parameter descriptions from
+    :returns: A tuple of (short_description, param_descriptions)
+    """
+    # Import here to avoid circular imports
+    from haystack.core import SuperComponent
+
+    # Get descriptions from the component's run method
+    short_desc, param_descriptions = _get_param_descriptions(component.run)
+
+    # If it's a SuperComponent, enhance the descriptions from the original components
+    if isinstance(component, SuperComponent):
```

Comment:
@vblagoje let's make the check for `_SuperComponent` otherwise this check will fail for super components that are made using the `@super_component` decorator like the `DocumentPreprocessor`